src/controller/CommissionerDiscoveryController.cpp

```diff
@@ -259,8 +455,25 @@ void CommissionerDiscoveryController::Ok()
     ChipLogDetail(Controller, "------Via Shell Enter: controller ux ok [passcode]");
 }
 
+/// Callback for getting execution into the main chip thread
+void CallbackCommissionWithPasscode(System::Layer * aSystemLayer, void * aAppState)
+{
+    uint32_t * tmpPasscode = static_cast<uint32_t *>(aAppState);
+    GetCommissionerDiscoveryController()->InternalCommissionWithPasscode(*tmpPasscode);
+    delete tmpPasscode;
+}
+
 void CommissionerDiscoveryController::CommissionWithPasscode(uint32_t passcode)
 {
+    uint32_t * tmpPasscode = new uint32_t(passcode);
+    DeviceLayer::SystemLayer().StartTimer(System::Clock::Seconds32(0), CallbackCommissionWithPasscode,
```

Comment:
Same comments as above.

Reply:
added return check and removed this temporary passcode object